ambry-api/src/main/java/com.github.ambry/rest/RestUtils.java

```diff
@@ -320,15 +321,6 @@ public static BlobProperties buildBlobProperties(Map<String, Object> args) throw
 
     long ttl = Utils.Infinite_Time;
     Long ttlFromHeader = getLongHeader(args, Headers.TTL, false);
-    if (container.isTtlRequired()) {
```

Comment:
I'm not clear on why you removed this section.  Don't we still need to check the header?

Reply:
I'v moved where the check is done because more business logic (and access to configuration) was required. The check (in spirit) has been moved to `PostBlobHandler`